### ISSUE TITLE
Report superceded/replaced drafts in search

### ISSUE THREAD

**Issue by ekr** [ietf-tools/datatracker#6276]
### Description

By default the datatracker doesn't show expired drafts. However, when a draft has been superceded by some other draft which is still active or an RFC I think we should show it by default.

Right now the behavior is inconsistent:

When I search for "rescorla" in datatracker, I get two RFCs: 5430 and 4347, both of which were individual submissions. 

However, there are many draft-rescorla drafts which were subsequently adopted and then became RFCs.

For instance:
https://datatracker.ietf.org/doc/draft-rescorla-tls-dtls13/, which then became draft-ietf-tls-dtls13 which then became RFC 9147

This seems like not a great state of affac

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by rjsparks** [comment]
That https://datatracker.ietf.org/doc/draft-rescorla-tls-dtls13/ doesn't show RFC9147 in the header graphic is unexpected, and whatever is at the root of that may also be affecting the search results.

Given that https://datatracker.ietf.org/doc/search?rfcs=on&activedrafts=on&by=author&author=rescorla returns such a wildly different result, this is likely not an issue with _just_ -tls-dtls13.

Once that's worked out, then at:
 
>perhaps by redirecting you to that descendant

We would be showing the descendants as the search results already (as we did for RFC5430 above. Are you proposing something different?

**Comment by ekr** [comment]
Yes, it's not not just a problem with dtls13. That was merely an example. For instance,

https://datatracker.ietf.org/doc/draft-thomson-ohai-ohttp/ behaves the same way (search for "thomson"). Note that in that case the superseding draft is live and there is no RFC.

I think what 5430 does is fine for things that became an RFC but note that you also need to handle the case of drafts that are now WG drafts, etc.

**Comment by larseggert** [comment]
The endpoint that creates `doc.json` (which the js that makes the timeline graph works on) fails to pick up the RFC as a related document.
